src/main/java/org/opensearch/agent/tools/VisualizationsTool.java

```diff
@@ -0,0 +1,176 @@
+        this.client = client;
+        this.index = index;
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery();
+        boolQueryBuilder.must().add(QueryBuilders.termQuery("type", SAVED_OBJECT_TYPE));
+        boolQueryBuilder.must().add(QueryBuilders.matchQuery(SAVED_OBJECT_TYPE + ".title", parameters.get("input")));
+
+        SearchSourceBuilder searchSourceBuilder = SearchSourceBuilder.searchSource().query(boolQueryBuilder);
+        searchSourceBuilder.from(0).size(3);
+        SearchRequest searchRequest = Requests.searchRequest(index).source(searchSourceBuilder);
+
+        client.search(searchRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(SearchResponse searchResponse) {
+                SearchHits hits = searchResponse.getHits();
+                StringBuilder visBuilder = new StringBuilder();
+                visBuilder.append("Title,Id\n");
+                if (hits.getTotalHits().value > 0) {
+                    Arrays.stream(hits.getHits()).forEach(h -> {
+                        String id = trimIdPrefix(h.getId());
+                        Map<String, String> visMap = (Map<String, String>) h.getSourceAsMap().get(SAVED_OBJECT_TYPE);
+                        String title = visMap.get("title");
+                        visBuilder.append(String.format(Locale.ROOT, "%s,%s\n", title, id));
+                    });
+
+                    listener.onResponse((T) visBuilder.toString());
+                } else {
+                    listener.onResponse((T) "No Visualization found");
+                }
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                if (ExceptionsHelper.unwrapCause(e) instanceof IndexNotFoundException) {
+                    listener.onResponse((T) "No Visualization found");
+                } else {
+                    listener.onFailure(e);
+                }
+            }
+        });
+    }
+
+    @VisibleForTesting
+    String trimIdPrefix(String id) {
+        id = Optional.ofNullable(id).orElse("");
+        if (id.startsWith(SAVED_OBJECT_TYPE)) {
+            String prefix = String.format(Locale.ROOT, "%s:", SAVED_OBJECT_TYPE);
+            return id.substring(prefix.length());
+        }
+        return id;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public String getVersion() {
+        return VERSION;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
```

Comment:
Most of the other tools are using the Lombok `@Getter` and `@Setter` annotation.  You'll need the getters for type and version I think, and maybe add annotation for description?  Look at the other tools for confirmation.

Reply:
thanks for suggestion, use lombok annotation now.